src/utils/add-due-date-comment.ts

```diff
@@ -0,0 +1,19 @@
+import { CommentList, SingleComment } from '../types/github';
+import { paginateAllCommentsOnIssue } from './paginate-comments-on-issue';
+import { context } from '@actions/github';
+import { octokit } from '../octokit';
+
+export const addDueDateComment = async (deadline: number, createdDate: Date, issue_number: number, numComments: number) => {
+  const hasExistingComments = numComments > 0;
+  const commentList: CommentList = hasExistingComments ? await paginateAllCommentsOnIssue(issue_number) : [];
+  // Create due date comment if there are no existing comments or the comment list does not contain a due date comment
+  if (!hasExistingComments || commentList.findIndex((comment: SingleComment) => comment.body?.startsWith('This issue is due on')) === -1) {
```

Comment:
`.find()` should be sufficient here
```suggestion
  if (!hasExistingComments || !commentList.find((comment: SingleComment) => comment.body?.startsWith('This issue is due on'))) {
```